### PR TITLE
Starting work to drop ComputeGrainCenterPostprocessor

### PR DESCRIPTION
**Pull request by permcody**
I may need a second pair of eyes (@SudiptaBiswas), but this is a long ways down the path of getting rid of the `ComputeGrainCenterUserObject`. I talked with @dschwen and it may be worthwhile to take some of the existing implementation and add it to the `FauxGrainTracker` if we want a simpler interface to maintain for some of these smaller test problems.

refs #7233

### CODE REVIEW

File: modules/phase_field/src/base/PhaseFieldApp.C [hunk 1]
```diff
@@ -409,7 +408,6 @@ PhaseFieldApp::registerObjects(Factory & factory)
   registerAux(OutputEulerAngles);
   registerAux(EulerAngleProvider2RGBAux);
 
-  registerUserObject(ComputeGrainCenterUserObject);
```
Comment (dschwen) [inline modules/phase_field/src/base/PhaseFieldApp.C]: This is not how we drop objects. Why don't you use registerDeprecated here?

File: modules/phase_field/src/materials/GrainAdvectionVelocity.C [hunk 2]
```diff
@@ -23,9 +24,7 @@ InputParameters validParams<GrainAdvectionVelocity>()
 
 GrainAdvectionVelocity::GrainAdvectionVelocity(const InputParameters & parameters) :
    DerivativeMaterialInterface<Material>(parameters),
-   _grain_data(getUserObject<ComputeGrainCenterUserObject>("grain_data")),
-   _grain_volumes(_grain_data.getGrainVolumes()),
-   _grain_centers(_grain_data.getGrainCenters()),
+   _grain_tracker(getUserObject<GrainTrackerInterface>("grain_data")),
```
Comment (dschwen) [inline modules/phase_field/src/materials/GrainAdvectionVelocity.C]: We should standardize the parameter name for `GrainTrackerInterface` type parameters.

Comment (permcody) [inline modules/phase_field/src/materials/GrainAdvectionVelocity.C]: agreed

File: modules/phase_field/src/base/PhaseFieldApp.C (hunk 1 above)
Comment (permcody) [inline modules/phase_field/src/base/PhaseFieldApp.C]: Right - it's only used in these test problems. It should probably just go.

**Comment by permcody** [comment]
@SudiptaBiswas is going to start from the work I've already put in and clean this up more. She's already asked me about changing the interface a bit more before we merge.

File: modules/phase_field/tests/rigidbodymotion/grain_forcesum.i [hunk 3]
```diff
@@ -171,30 +171,30 @@
...
+#    index = 1
+#    property = advection_velocity
+#  [../]
+#  [./vadv11]
+#    type = MaterialStdVectorRealGradientAux
+#    variable = vadv11
+#    property = advection_velocity
+#    index = 1
+#    component = 1
+#  [../]
```
Comment (dschwen) [inline modules/phase_field/tests/rigidbodymotion/grain_forcesum.i]: Uhm, are we still testing anything here?!

**Comment by dschwen** [comment]
Does any of the rigid body motion now still work? Don't we need to make the Kernels GT aware now as well? Everything is now stored in terms of grain numbers rather than OP numbers. The mapping will probably not be 1:1 (especially since this does not add `FauxGrainTracker` support.

**Comment by permcody** [comment]
We certainly want the rigid body motion to work. I'll add the FauxGrainTracker support but I'll rely on @SudiptaBiswas to help make the magic of the new number work properly in all of those examples. At this point, I think the GrainGrowth usage of GrainTracker is pretty solid. It's time to start working on new phase-field models like sintering and nucleation.

**Comment by SudiptaBiswas** [comment]
I had a brief discussion with @permcody on the rigid body motion tests. I will look into the tests and modify them as required. I will let you know if I need any additional capability to make it work.

**Comment by SudiptaBiswas** [comment]
Okay, here are some thoughts
1. For rigid body motions, getting grain centers corresponding to grain numbers will work only if each grain no. is associated with a single particle (that was identified with individual order parameters earlier).
2. If we have multi-grain particle (referring to @tonkmr's new IC), we will need center of each particles and not grains for this. Do we have any other way to obtain grain centers based on order parameters? 
3. Now, if multiple particles are identified with same order parameter and have multiple grains in side them then it's bit more complicated I guess. We will need to think about how to identify each grains and particles seperately. 
4. I agree with @dschwen, we will need the RigidBodyMotion kernels to communicate with grain tracker. I am not sure if @itgreenquist is already on it or have plans. 

Let me know if it's not clear what I meant. Having said that, I know GrainCenterUserObject is not used much, but I would appreciate if we can deprecate this object and keep it hanging longer. This might help in comparing/debugging while switching other RBM objects in near future.

**Comment by permcody** [comment]
I can't answer all these questions because I don't know the area all that
well. I am definitely willing to help out and build the capabilities needed
though. I do have one insight though, it appears that we basically have two
different types of models: those that have order parameters == num grains
and those that have order parameters < num grains. The interface for
working with those two should really be the same. I propose that we simply
move the existing centers and volume logic into "FauxGrainTracker" then all
these simulations will be ready to scale up with a consistent interface.
I'm not attached to the FauxGrainTracker name btw, we can certainly rename
it.

**Comment by itgreenquist** [comment]
On @SudiptaBiswas's thoughts:
1. I agree. The getElementalValues function in GrainTracker seems to be the easiest way to do that.
2. This really complicates things, but I am not convinced that will be necessary. The free energy may be enough to hold the particles together in the simulations. After all, it is what will hold the grains together in a single particle after sintering. Plus the change in force across a particle will hopefully not be so extreme to tear the particle apart. I think we should try it with centers for each grain first, and check to see if centers for each particle are really necessary.
3. If I am wrong and we do need centers for each particle, I think another layer of order parameters may be the easiest solution. The hardest part of that would probably be figuring out how to integrate them into the free energy function.
4. I haven't considered this before and I am not sure I understand. Would the idea be to make kernels for each grain number instead of for each op number? Wouldn't that defeat the purpose of using GrainTracker?

On @permcody's thought:
 I agree that the interface should be the same and I like that plan.

**Comment by permcody** [comment]
On Mon, Jul 18, 2016 at 8:51 AM itgreenquist <notifications@github.com>
wrote:

> On @SudiptaBiswas <https://github.com/SudiptaBiswas>'s thoughts:
> 1. I agree. The getElementalValues function in GrainTracker seems to be
> the easiest way to do that.
>

Actually probably not. That method is for finding all contributing grains
typically in the grain boundary region. If you just want grain centers and
grain volumes, we are creating a new interface for that.


> 2. This really complicates things, but I am not convinced that will be
> necessary. The free energy may be enough to hold the particles together in
> the simulations. After all, it is what will hold the grains together in a
> single particle after sintering. Plus the change in force across a particle
> will hopefully not be so extreme to tear the particle apart. I think we
> should try it with centers for each grain first, and check to see if
> centers for each particle are really necessary.
>
This is probably something that I'll work on. I'll talk to Mike Tonks a bit
more about it.

**Comment by SudiptaBiswas** [comment]
Well, theoretically all grains inside a particle should move with same velocity as a single rigid body. Also the force responsible for RBM is generated when two particles are in contact, it does not exist on all the grain boundaries. For multigrain-particles we probably will need two sets of variables, nonlinear variables for particles and aux for grains.

**Comment by SudiptaBiswas** [comment]
I believe the first step here should be to make the existing RBM objects work with grain tracker. We can worry about multigrain-particles IC later. @itgreenquist are you working on it? If not I can start on that.

**Comment by itgreenquist** [comment]
I will start working on the RBM objects as soon as this pull request goes through.

**Comment by SudiptaBiswas** [comment]
This PR needs some work on the RBM objects as well. I will take care of that then.

**Comment by permcody** [comment]
How do we want to handle the centroids of grains spanning periodic boundaries? I ask because if you want a single centroid for a grain this is actually non-trivial when PBCs are used. I have addressed the bounding box problem by keeping multiple bounding boxes (one per grain piece) around in the data structure.

**Comment by itgreenquist** [comment]
I haven't really looked at how GrainTracker deals with periodic boundaries as is, so I don't think I can be much help on coding. However, I do not have plans for using periodic boundary conditions with RBM, and I have a hard time imagining a scenario where periodic boundaries would be helpful in sintering and densification models. So I would be fine if we didn't address periodic boundaries. I don't know if @SudiptaBiswas agrees or if you just want to go ahead and get it done though...

**Comment by permcody** [comment]
The volume calculation doesn't change with PBCs. The centroid calculation
however does. Right now the centroid is just plain wrong if PBCs are used.
I know it can be fixed... I'll put it on the back burner unless somebody
needs it right away.

**Comment by SudiptaBiswas** [comment]
I understand obtaining center for a particle when it's spanned across the boundary would be tricky. I do not use PBCs much either with sintering. Even if I need PBC for some testing, I can definitely avoid having particles across the boundary.

**Comment by permcody** [comment]
Superceeded by #7533

**Pull request closed without merging**